src/cluster.cc

```diff
@@ -53,6 +54,26 @@ Status Cluster::SetNodeId(std::string node_id) {
   return Status::OK();
 }
 
+Status Cluster::SetSlot(int slot, std::string node_id) {
+  if (slot < 0 || slot >= HASH_SLOTS_SIZE) {
+    return Status(Status::NotOK, "Slot is out of range");
+  }
+  if (node_id.empty() || nodes_.find(node_id) == nodes_.end()) {
+    return Status(Status::NotOK, "Invalid node id");
+  }
+
+  std::lock_guard<std::mutex> guard(cluster_mutex_);
+  if (slots_nodes_[slot]) {
+    slots_nodes_[slot]->slots_.set(slot, 0);
+  }
+  nodes_[node_id]->slots_.set(slot, 1);
+  slots_nodes_[slot] = nodes_[node_id];
+
+  // Mark topology changed
+  version_++;
```

Comment:
maybe it is not a good idea to change cluster topology and version_ directly, since we use controler to manage everything,
An idea: we set the slot to `imported` state on the dst node which can serve requests on the imported slot in this status, and we set the slot to `migrated` state on the src node which will `MOVE` the requests on the migrated slot,